[BUG] REPRAP_DISCOUNT_SMART_CONTROLLER and ROTATE_PROGRESS_DISPLAY fails to compile

### Bug Description

Using the latest `bugfix-2.0.x`(41529b6), `REPRAP_DISCOUNT_SMART_CONTROLLER` with `ROTATE_PROGRESS_DISPLAY` causes the build to fail. This was due to the "Reduce, reuse, recycle" (b6bb4fa) refactoring commit in https://github.com/MarlinFirmware/Marlin/pull/20148 before it was merged.

<details><summary>Error output:</summary>
<p>

```prolog
Marlin/src/lcd/HD44780/marlinui_HD44780.cpp: In function 'uint8_t draw_elapsed_or_remaining_time(uint8_t)':
Marlin/src/lcd/HD44780/marlinui_HD44780.cpp:714:61: error: 'blink' was not declared in this scope
  714 |     const bool show_remain = TERN1(ROTATE_PROGRESS_DISPLAY, blink) && (printingIsActive() || marlin_state == MF_SD_COMPLETE);
      |                                                             ^~~~~
Marlin/src/lcd/HD44780/../../inc/../core/macros.h:419:26: note: in definition of macro 'THIRD'
  419 | #define THIRD(a,b,c,...) c
      |                          ^
Marlin/src/lcd/HD44780/../../inc/../core/macros.h:196:29: note: in expansion of macro '___TERN'
  196 | #define __TERN(T,V...)      ___TERN(_CAT(_NO,T),V)  // Prepend '_NO' to get '_NOT_0' or '_NOT_1'
      |                             ^~~~~~~
Marlin/src/lcd/HD44780/../../inc/../core/macros.h:195:29: note: in expansion of macro '__TERN'
  195 | #define _TERN(E,V...)       __TERN(_CAT(T_,E),V)    // Prepend 'T_' to get 'T_0' or 'T_1'
      |                             ^~~~~~
Marlin/src/lcd/HD44780/../../inc/../core/macros.h:193:29: note: in expansion of macro '_TERN'
  193 | #define TERN1(O,A)          _TERN(_ENA_1(O),1,A)    // OPTION converted to A or '1'
      |                             ^~~~~
Marlin/src/lcd/HD44780/marlinui_HD44780.cpp:714:30: note: in expansion of macro 'TERN1'
  714 |     const bool show_remain = TERN1(ROTATE_PROGRESS_DISPLAY, blink) && (printingIsActive() || marlin_state == MF_SD_COMPLETE);
      |                              ^~~~~
Compiling .pio/build/BIGTREE_BTT002/src/src/libs/buzzer.cpp.o
Marlin/src/lcd/HD44780/marlinui_HD44780.cpp:719:28: error: 'get_progress_percent' was not declared in this scope
  719 |         uint8_t progress = get_progress_percent();
      |                            ^~~~~~~~~~~~~~~~~~~~
Compiling .pio/build/BIGTREE_BTT002/src/src/libs/crc16.cpp.o
Compiling .pio/build/BIGTREE_BTT002/src/src/libs/heatshrink/heatshrink_decoder.cpp.o
Compiling .pio/build/BIGTREE_BTT002/src/src/libs/nozzle.cpp.o
*** [.pio/build/BIGTREE_BTT002/src/src/lcd/HD44780/marlinui_HD44780.cpp.o] Error 1
```

</p>
</details>

### Configuration Files

https://github.com/thisiskeithb/Marlin/tree/myconfig/MK3S-BTT002-TMC2209

### Steps to Reproduce

1. Use config from above or enable `REPRAP_DISCOUNT_SMART_CONTROLLER` with `ROTATE_PROGRESS_DISPLAY`.
2. Attempt to build.
3. See build fail.

**Expected behavior:**

Build not to fail.

**Actual behavior:**

Build fails.

--- a/Marlin/src/lcd/HD44780/marlinui_HD44780.cpp
+++ b/Marlin/src/lcd/HD44780/marlinui_HD44780.cpp
@@ -707,8 +707,7 @@ void MarlinUI::draw_status_message(const bool blink) {
  *  |01234567890123456789|
  */
 
-inline uint8_t draw_elapsed_or_remaining_time(uint8_t timepos) {
-  const bool blink = ui.get_blink();
+inline uint8_t draw_elapsed_or_remaining_time(uint8_t timepos, const bool blink) {
   char buffer[14];
 
   #if ENABLED(SHOW_REMAINING_TIME)
@@ -875,7 +874,7 @@ void MarlinUI::draw_status_screen() {
       lcd_put_u8str(i16tostr3rj(feedrate_percentage));
       lcd_put_wchar('%');
 
-      const uint8_t timepos = draw_elapsed_or_remaining_time(LCD_WIDTH - 1);
+      const uint8_t timepos = draw_elapsed_or_remaining_time(LCD_WIDTH - 1, blink);
 
       #if LCD_WIDTH >= 20
         lcd_moveto(timepos - 7, 2);
@@ -974,7 +973,7 @@ void MarlinUI::draw_status_screen() {
 
     #else
 
-      (void)draw_elapsed_or_remaining_time(LCD_WIDTH - 4);
+      (void)draw_elapsed_or_remaining_time(LCD_WIDTH - 4, blink);
 
     #endif
 

--- a/Marlin/src/lcd/HD44780/marlinui_HD44780.cpp
+++ b/Marlin/src/lcd/HD44780/marlinui_HD44780.cpp
@@ -708,6 +708,7 @@ void MarlinUI::draw_status_message(const bool blink) {
  */
 
 inline uint8_t draw_elapsed_or_remaining_time(uint8_t timepos) {
+  const bool blink = ui.get_blink();
   char buffer[14];
 
   #if ENABLED(SHOW_REMAINING_TIME)
@@ -716,7 +717,7 @@ inline uint8_t draw_elapsed_or_remaining_time(uint8_t timepos) {
       #if ENABLED(USE_M73_REMAINING_TIME)
         duration_t remaining = get_remaining_time();
       #else
-        uint8_t progress = get_progress_percent();
+        uint8_t progress = ui.get_progress_percent();
         uint32_t elapsed = print_job_timer.duration();
         duration_t remaining = (progress > 0) ? ((elapsed * 25600 / progress) >> 8) - elapsed : 0;
       #endif

